vertx-service-discovery/src/main/java/io/vertx/servicediscovery/impl/DefaultServiceDiscoveryBackend.java

```diff
@@ -66,57 +94,84 @@ public void remove(Record record, Handler<AsyncResult<Record>> resultHandler) {
   @Override
   public void remove(String uuid, Handler<AsyncResult<Record>> resultHandler) {
     Objects.requireNonNull(uuid, "No registration id in the record");
-    registry.remove(uuid, ar -> {
-      if (ar.succeeded()) {
-        if (ar.result() == null) {
-          // Not found
-          resultHandler.handle(Future.failedFuture("Record '" + uuid + "' not found"));
+    retrieveRegistry(registry -> {
+        if (registry.failed()) {
+          resultHandler.handle(Future.failedFuture("Unable to retrieve the registry"));
```

Comment:
Same here (re losing the cause)

Reply:
I don't really care of the cause here. But ok.

Reply:
As a user you want to know why you couldn't retrieve the registry. Would it be only to write a proper bug report :)